src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -409,6 +409,18 @@ private void InitNativeProcess()
             // Get the start info for the process.
             ProcessStartInfo startInfo = GetProcessStartInfo(redirectOutput, redirectError, redirectInput, soloCommand);
 
+#if !UNIX
+            if (this.Path.ToLowerInvariant().EndsWith("powershell.exe") || this.Path.ToLowerInvariant().EndsWith("powershell_ise.exe"))
```

Comment:
If you assign `this.Path.ToLowerInvariant()` to a variable, we can avoid a second culture lookup/allocation.

Reply:
Do we think anyone starts `powershell.exe` etc with `Start-Process` or some other mechanism not covered here where it would presumably inherit the PSModulePath?

Reply:
will update.  I don't think using `Start-Process` is a common case, we can address that if users raise an issue